Update quarkus version checks to adapt to new SNAPSHOT versions

zakkak opened the pull request:
See
https://github.com/quarkusio/quarkus/blob/3c8c772dda9ed99674c9d8a6cd9cb4635d1ce5a2/build-parent/pom.xml#L8
and
https://github.com/quarkusio/quarkus/blob/bc0ca288d32ba2df212d8c4a2783c0bb21ca99a1/build-parent/pom.xml#L8

Relates to https://github.com/graalvm/mandrel/issues/501

jerboaa commented on testsuite/src/it/java/org/graalvm/tests/integration/PerfCheckTest.java:
```diff
@@ -139,7 +139,7 @@ public void testQuarkusJSONParseOnce(TestInfo testInfo) throws IOException, Inte
             assertTrue(app.buildAndRunCmds.cmds.length > 1);
 
-            if (QUARKUS_VERSION.majorIs(3) || QUARKUS_VERSION.isSnapshot()) {
+            if (QUARKUS_VERSION.compareTo(new QuarkusVersion("3.0.0")) >= 0) {
```
Consider extracting the `3.0.0` version into a constant.

zakkak replied:
Done

jerboaa reviewed (approved):
Consider adding some tests for `QuarkusVersion` comparisons so that this keeps working. Looks fine to me, though.

zakkak commented:
> Consider adding some tests for QuarkusVersion comparisons so that this keeps working. Looks fine to me, though.

I extracted constant versions in constant fields in QuarkusVersion and also introduced tests for the class.

Thanks for the review @jerboaa

jerboaa commented:
> I extracted constant versions in constant fields in QuarkusVersion and also introduced tests for the class.

Very nice, thank you!
